sphinx/source/performance.rst

```diff
@@ -0,0 +1,103 @@
+    ===============================================================================
+    hash:
+    ===============================================================================
+       Name (baseline is *)   |   Dim   |  Total ms |  ns/op  |Baseline| Ops/second
+    ===============================================================================
+               hash_384_1byte |       1 |     0.001 |     600 |  0.750 |  1666666.7
+        hash_256k1_mbed_1byte |       1 |     0.001 |     600 |  0.750 |  1666666.7
+      hash_256k1_bitc_1byte * |       1 |     0.001 |     800 |      - |  1250000.0
+                  hash_384_1k |       1 |     0.003 |    3100 |  3.875 |   322580.6
+           hash_256k1_mbed_1k |       1 |     0.004 |    4200 |  5.250 |   238095.2
+         hash_256k1_bitc_1k * |       1 |     0.004 |    4500 |  5.625 |   222222.2
+                hash_384_100k |       1 |     0.223 |  223000 |278.750 |     4484.3
+         hash_256k1_mbed_100k |       1 |     0.340 |  340100 |425.125 |     2940.3
+       hash_256k1_bitc_100k * |       1 |     0.340 |  340400 |425.500 |     2937.7
+    ===============================================================================
+
+This compares the signing, verification, and hashing performance of different cryptographic curve implementations in CCF. The `bitcoin-core implementation <bitcoin_256k1>`_ of elliptic curve secp256k1 is significantly faster than other supported curves, particularly when verification becomes a bottleneck.
+
+End-to-end performance tests
+----------------------------
+
+The end-to-end service performance tests can also be from the CCF build directory:
+
+.. code-block:: bash
+
+    ./tests.sh -VV -L "perf"
+
+Each of these tests creates a temporary CCF service on the local machine, then sends a high volume of transactions to measure peak and average throughput. The python test wrappers will print summary statistics including a transaction rate histogram when the test completes. These statistics can be retrieved from any CCF service via the ``getMetrics`` RPC.
+
+For a finer grained view of performance the clients in these tests can also dump the precise times each transaction was sent and its response received, for later analysis. The ``samples`` folder contains a ``plot_tx_times`` Python script demonstrating how this data may be plotted. The following plot was produced by this script, showing 1,000 transactions sent to the `SmallBank`_ sample application:
+
+.. image:: img/1k_unsigned.png
+
+* green points are SmallBank business transactions
+* red points are generic ``getCommit`` requests used to poll for global commit synchronisation
+* blue line shows the receiving node's local commit id
+* orange line shows the service's global commit id
+
+This shows a healthy service, with minimal latency responses to each request, write transactions causing an increment to the local commit id, and only a few slower transactions around global commit changes.
```

Comment:
"and only a few slower transactions around global commit changes." -> I'm not sure what you mean here? Also, could we explain why the global commit increases in big steps?